### PR TITLE
Switch code signing to use xcconfig files

### PR DESCRIPTION
**Pull request by FahimF**
* Switched code signing over to use xcconfig files so that multiple people can work on the project easily
* Refactored some of the code for naming consistency and to separate out functionality

### CODE REVIEW

File: Diffusion.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -502,11 +524,12 @@
...
 				MARKETING_VERSION = 1.0;
-				PRODUCT_BUNDLE_IDENTIFIER = com.pcuenca.Diffusion;
+				PRODUCT_BUNDLE_IDENTIFIER = "$(inherited)";
 				PRODUCT_NAME = "$(TARGET_NAME)";
+				PROVISIONING_PROFILE_SPECIFIER = "$(inherited)";
 				SDKROOT = auto;
-				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
-				SUPPORTS_MACCATALYST = YES;
+				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator macosx";
+				SUPPORTS_MACCATALYST = NO;
```
Comment (pcuenca) [inline Diffusion.xcodeproj/project.pbxproj:532]:
```suggestion
				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
				SUPPORTS_MACCATALYST = YES;
```

Can we remove this from this PR? Happy to support native macOS UI when we need it, but unrelated to xcconfig files.

File: Diffusion.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -542,11 +567,12 @@
...
```
Comment (pcuenca) [inline Diffusion.xcodeproj/project.pbxproj:575]:
```suggestion
				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
				SUPPORTS_MACCATALYST = YES;
```

File: Diffusion/Pipeline/Pipeline.swift [hunk 3]
```diff
@@ -29,15 +29,16 @@ class Pipeline {
         self.pipeline = pipeline
     }
     
-    func generate(prompt: String, scheduler: StableDiffusionScheduler, numInferenceSteps stepCount: Int = 50, seed: UInt32? = nil) throws -> CGImage {
+	func generate(prompt: String, scheduler: StableDiffusionScheduler, numInferenceSteps stepCount: Int = 50, imageCount: Int = 1, safetyOn: Bool = false, seed: UInt32? = nil) throws -> CGImage {
```
Comment (pcuenca) [inline Diffusion/Pipeline/Pipeline.swift:32]: Can we revert the UI changes and additional options from this PR? They are very useful, but I think it's better to deal with them separately if that's possible.

File: Diffusion/Views/LoadingView.swift [hunk 4]
```diff
@@ -33,7 +33,7 @@ struct LoadingView: View {
     var body: some View {
         VStack {
             switch currentView {
-            case .textToImage: TextToImage().transition(.opacity)
+            case .textToImage: TextToImageView().transition(.opacity)
```
Comment (pcuenca) [inline Diffusion/Views/LoadingView.swift:36]: Same here, let's try to do the renames separately.

File: Diffusion/Views/PreviewView.swift [hunk 5]
```diff
@@ -0,0 +1,46 @@
+//
```
Comment (pcuenca) [inline Diffusion/Views/PreviewView.swift:1]: Same here, to be added later.

**Review by pcuenca — COMMENTED** [review]
Thanks a lot, I think using configuration files will be useful for anyone wanting to try the repo themselves, but I think it's better to keep the PR focused on that. If possible, let's undo the UI and additional options and deal with them in upcoming PRs. 🙏

**Comment by pcuenca** [comment]
Hi @FahimF! I just submitted a extremely simplified version of this PR (and added you as an author) in which:
- Code signing for macOS is disabled by default, to address #5.
- Users need to update their team id in a single `common.xcconfig` file.

If we need to make it more complicated, then we can go the route of creating a template and adding the configuration files to `.gitignore`, but I think this could suffice for now.

Would it be ok to close this PR and work on the other improvements separately?

Thanks a lot!

**Comment by FahimF** [comment]
Sorry @pcuenca been really busy with other stuff and haven't had a chance to work on this. Yes, I'll close this PR and we can look at some of the other stuff later ....

**Comment by pcuenca** [comment]
No worries at all!

**Pull request closed without merging**